pkg/resources/management.cattle.io/v3/project/validator.go

```diff
@@ -177,7 +179,18 @@ func usedQuotaFits(usedQuota, projectQuota *v3.ResourceQuotaLimit) (*field.Error
 	}
 	fits, exceeded := quotaFits(usedQuotaResourceList, projectQuotaResourceList)
 	if !fits {
-		return field.Forbidden(projectSpecFieldPath.Child(projectQuotaField), fmt.Sprintf("resourceQuota is below the used limit on fields: %s", format.ResourceList(exceeded))), nil
+		return field.Forbidden(projectSpecFieldPath.Child(projectQuotaField), fmt.Sprintf("resourceQuota is below the used limit on fields: %s", formatResourceList(exceeded))), nil
 	}
 	return nil, nil
 }
+
+// directly copied from https://github.com/kubernetes/kubernetes/blob/a66aad2d80dacc70025f95a8f97d2549ebd3208c/pkg/kubelet/util/format/resources.go
+func formatResourceList(resources v1.ResourceList) string {
```

Comment:
Missed the k8s.io/kubernetes upgrade somehow. The `format.ResourceList` function was removed some time ago in the upstream project. I'm not sure what's the policy on copying (keep attribution somehow, etc etc?) and also I see that the rancher/rancher project is taking the same approach here: https://github.com/rancher/rancher/pull/42691. So I'm wondering, where should that function live? Do we want a common place for it or do we keep it copied in both?

Reply:
This function is straightforward and has no critical usage, so I think we can follow the go proverb.
> [A little copying is better than a little dependency.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s)